include/circt/Dialect/ESI/ESIInterfaces.td

```diff
@@ -0,0 +1,49 @@
+//===- ESIInterfaces.td - ESI Interfaces -----------------*- tablegen -*---===//
+//
+// Part of the LLVM Project, under the Apache License v2.0 with LLVM Exceptions.
+// See https://llvm.org/LICENSE.txt for license information.
+// SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+//
+//===----------------------------------------------------------------------===//
+//
+// This describes the interfaces in the ESI dialect.
+//
+//===----------------------------------------------------------------------===//
+
+include "mlir/IR/OpBase.td"
+
+def ChannelOpInterface : OpInterface<"ChannelOpInterface"> {
+  let description =
+      [{"An interface for operations which carries channel semantics."}];
+
+  let methods = [
+    InterfaceMethod<
+      [{"Returns any value of this operation that carries data semantics.
+         This value is only used for inspection purposes and may therefore point
+         to either in or outputs of an operation."}],
+      "mlir::Value", "anyChannel"
```

Comment:
Is this used anywhere which actually needs a `Value`? If not, could it return the `innerType` and be called something like that?

Reply:
+1 - initially did it for value to reduce typing at the implementation sites. But it agree, passing random values around doesn't make much sense, since all we care about for this interface is the underlying type.